src/api/objects/ObjectAPI.js

```diff
@@ -215,6 +227,188 @@ define([
         return utils.makeKeyString(identifier);
     };
 
+    let migrations = [
+        {
+            check(domainObject) {
+                return domainObject.type === 'layout' && domainObject.configuration.layout;
+            },
+            migrate(domainObject) {
+                let childObjects = {};
+                let promises = Object.keys(domainObject.configuration.layout.panels).map(key => {
+                    return openmct.objects.get(key)
+                        .then(object => {
+                            childObjects[key] = object;
+                        });
+                });
+
+                return Promise.all(promises)
+                    .then(function () {
+                        return migrateDisplayLayout(domainObject, childObjects);
+                    });
+            },
+        },
+        {
+            check(domainObject) {
+                return domainObject.type === 'telemetry.fixed' && domainObject.configuration['fixed-display'];
+            },
+            migrate(domainObject) {
+                const DEFAULT_GRID_SIZE = [64, 16];
+                let newLayoutObject = {
+                    identifier: domainObject.identifier,
+                    location: domainObject.location,
+                    name: domainObject.name,
+                    type: "layout"
+                };
+                let layoutType = openmct.types.get('layout');
+                layoutType.definition.initialize(newLayoutObject);
+                newLayoutObject.composition = domainObject.composition;
+                newLayoutObject.configuration.layoutGrid = domainObject.layoutGrid || DEFAULT_GRID_SIZE;
+
+                let elements = domainObject.configuration['fixed-display'].elements;
+                let telemetryObjects = {};
+                let promises = elements.map(element => {
+                    if (element.id) {
+                        return openmct.objects.get(element.id)
+                            .then(object => {
+                                telemetryObjects[element.id] = object;
+                            });
+                    }
+                });
+
+                return Promise.all(promises)
+                    .then(function () {
+                        newLayoutObject.configuration.items =
+                            migrateFixedPositionConfigurataion(elements, telemetryObjects);
+                        return newLayoutObject;
+                    });
+            }
+        }
+    ];
+
+    function needsMigration(domainObject) {
+        return migrations.some(m => m.check(domainObject));
+    }
+
+    function migrateObject(domainObject) {
+        return migrations.filter(m => m.check(domainObject))
+            .reduce((o, m) => "").migrate(domainObject);
```

Comment:
Could you help me understand the role of 'reduce' here? I would expect the filter on migrations to only return one migration, in which case couldn't we just use [0]?

Reply:
You are right, reduce is unnecessary here.